-arch i386 should be removed for Darwin

Looks like we need to remove support for i386 architecture for newer OSX versions (ifeq ($(YQ2_OSTYPE), Darwin) case). There should be no "-arch i386" or "make" will fail with:

_ld: symbol(s) not found for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [release/game.dylib] Error 1
make: *** [3zb2] Error 2_

--- a/Makefile
+++ b/Makefile
@@ -84,7 +84,7 @@ endif
 # -MMD to generate header dependencies.
 ifeq ($(YQ2_OSTYPE), Darwin)
 CFLAGS := -O2 -fno-strict-aliasing -fomit-frame-pointer \
-		  -Wall -pipe -g -fwrapv -arch i386 -arch x86_64
+		  -Wall -pipe -g -fwrapv -arch arm64 -arch x86_64
 else
 CFLAGS := -O2 -fno-strict-aliasing -fomit-frame-pointer \
 		  -Wall -pipe -g -MMD -fwrapv
@@ -125,7 +125,7 @@ CFLAGS += -DOSTYPE=\"$(YQ2_OSTYPE)\" -DARCH=\"$(YQ2_ARCH)\"
 
 # Base LDFLAGS.
 ifeq ($(YQ2_OSTYPE), Darwin)
-LDFLAGS := -shared -arch i386 -arch x86_64 
+LDFLAGS := -shared -arch arm64 -arch x86_64 
 else ifeq ($(YQ2_OSTYPE), Windows)
 LDFLAGS := -shared -static-libgcc
 else

